modules/gdscript/language_server/gdscript_text_document.cpp

```diff
@@ -346,6 +346,16 @@ Dictionary GDScriptTextDocument::resolve(const Dictionary &p_params) {
 		}
 	}
 
+	if (item.kind == lsp::CompletionItemKind::Method) {
+		bool isTriggerCharacter = params.context.triggerKind == lsp::CompletionTriggerKind::TriggerCharacter;
+		bool isQuoteCharacter = params.context.triggerCharacter == "\"" || params.context.triggerCharacter == "'";
+
+		if (isTriggerCharacter && isQuoteCharacter && item.insertText.is_quoted()) {
+			item.insertText = item.insertText.lstrip("'").rstrip("'");
+			item.insertText = item.insertText.lstrip("\"").rstrip("\"");
```

Comment:
Wouldn't `item.insertText.unquote()` be equivalent / sufficient here?

To be clear that's the implementation:
```
String String::unquote() const {
        if (!is_quoted()) {
                return *this;
        }

        return substr(1, length() - 2);
}
```

So it just removes the first and last char after confirming those are quotes.

The different with your code is that it would also remove nested quotes of this form: `'"some text"'` (single quote then double quote). But I'm not sure that's wanted?

Reply:
Thank you for pointing out `unquote()`. I updated the code to use it, but left in the `is_quoted()` check for the reasons you mentioned.